scripts/o3de/tests/ExportScripts/test_export_source_built_project.py

```diff
@@ -0,0 +1,1351 @@
+    pytest.param(False, True, False, True),
+    pytest.param(False, True, False, False),
+    pytest.param(False, False, True, True),
+    pytest.param(False, False, True, False),
+    pytest.param(False, False, False, True),
+    pytest.param(False, False, False, False),
+
+    pytest.param(True, True, True, True),
+    pytest.param(True, True, True, False),
+    pytest.param(True, True, False, True),
+    pytest.param(True, True, False, False),
+    pytest.param(True, False, True, True),
+    pytest.param(True, False, True, False),
+    pytest.param(True, False, False, True),
+    pytest.param(True, False, False, False),
+])
+def test_build_game_targets_combinations(tmp_path, is_engine_centric, use_sdk, has_monolithic, use_monolithic):
+    test_project_name = "TestProject"
+    test_project_path = tmp_path / "project"
+    test_engine_path = tmp_path / "engine"
+
+    test_o3de_base_path = test_project_path if not is_engine_centric else test_engine_path
+    test_relative_base_path = pathlib.PurePath('this/is/relative')
+    test_absolute_base_path  = tmp_path / "other"
+
+    assert test_absolute_base_path.is_absolute()
+
+    test_output_path = tmp_path / "output"
+
+    mock_logger = create_autospec(logging.Logger)
+
+    with patch('o3de.manifest.is_sdk_engine', return_value=use_sdk) as mock_is_sdk_engine,\
+         patch('o3de.export_project.has_monolithic_artifacts', return_value=has_monolithic) as mock_has_mono_artifacts,\
+         patch('o3de.export_project.get_platform_installer_folder_name', return_value="Windows") as mock_platform_folder_name,\
+         patch('o3de.export_project.validate_project_artifact_paths', return_value=[]) as mock_validate_project_artifacts,\
+         patch('o3de.export_project.kill_existing_processes') as mock_kill_processes,\
+         patch('o3de.export_project.build_export_toolchain') as mock_build_export_toolchain,\
+         patch('o3de.export_project.build_game_targets') as mock_build_game_targets,\
+         patch('o3de.export_project.get_asset_processor_batch_path') as mock_get_asset_processor_path,\
+         patch('o3de.export_project.build_assets') as mock_build_assets,\
+         patch('o3de.export_project.get_asset_bundler_batch_path') as mock_get_asset_bundler_path,\
+         patch('o3de.export_project.bundle_assets') as mock_bundle_assets,\
+         patch('o3de.export_project.setup_launcher_layout_directory') as mock_setup_launcher_layout_directory,\
+         patch('logging.getLogger', return_value=mock_logger) as mock_get_logger:
+        
+
+        mock_ctx = create_autospec(O3DEScriptExportContext)
+        mock_ctx.project_path = test_project_path
+        mock_ctx.engine_path = test_engine_path
+        mock_ctx.project_name = test_project_name
+
+        mock_platform = 'pc'
+
+        for should_build_tools, toolconf in zip([True, False, True, False],
+                                                ['profile', 'profile', 'release', 'release']):
```

Comment:
are the build tools always "profile" I think only the game launcher can be "release"